core/test_utils/CMakeLists.txt

```diff
@@ -0,0 +1,34 @@
+include(FetchContent)
+include(GoogleTest)
+
+project(n_e_s_core_testutils)
+
+FetchContent_Declare(
```

Comment:
Should we put all third party dependencies is a new folder by the way and only declare them at one place?

Reply:
Yes, please. 😁 This was fine back when we only needed the dependency in 1 lib, but it's starting to get a bit messy now.
`third_party/libname`?